types/json/binary_functions.go

```diff
@@ -763,6 +763,59 @@ func MergeBinary(bjs []BinaryJSON) BinaryJSON {
 	return mergeBinaryArray(results)
 }
 
+// MergePatchBinary behaves the same as that of MergeBinaryPreserve, with the following two exceptions:
+// 1) MergePatchBinary removes any member in the first object with a matching key in the second object, provided that the value associated with the key in the second object is not JSON null.
+// 2) If the second object has a member with a key matching a member in the first object, MergePatchBinary replaces the value in the first object with the value in the second object, whereas MergeBinaryPreserve appends the second value to the first value.
+func MergePatchBinary(bjs []BinaryJSON) BinaryJSON {
+	var result = bjs[0]
+	for i := 1; i < len(bjs); i++ {
+		result = mergePatchBinaryObject(&result, &(bjs[i]))
+	}
+	return result
+}
+
+func mergePatchBinaryObject(oldObj *BinaryJSON, newObj *BinaryJSON) BinaryJSON {
+	if newObj.TypeCode != TypeCodeObject {
+		return *newObj
+	}
```

Comment:
I feel confused about this logic. The `MergeJson` function in golang would merge the multiple arrays into one array.